### DISCUSSION TITLE
failed to solve: process "/bin/sh -c apk --no-cache add curl &&     npm ci" did not complete successfully: exit code: 1

### DISCUSSION THREAD

**Discussion by XMLC2024** [danny-avila/LibreChat#1846] [Troubleshooting]
### What is your question?

The steps in the video are followed one by one.
I tried again many times, but the error message was the same every time.
The problems always appear in the last step.
Trying to run it as administrator doesn't work either
I tried it twice in total.
The first time I connected to VPN, it didn't work.
The second time I disconnected the VPN, it still didn’t work.
Console details below

### More Details

FIRST TRY
=======================================================
C:\>git clone https://github.com/danny-avila/LibreChat
Cloning into 'LibreChat'...
remote: Enumerating objects: 21901, done.
remote: Counting objects: 100% (773/773), done.
remote: Compressing objects: 100% (524/524), done.
remote: Total 21901 (delta 342), reused 513 (delta 239), pack-reused 21128
Receiving objects: 100% (21901/21901), 23.57 MiB | 8.81 MiB/s, done.
Resolving deltas: 100% (15564/15564), done.

C:\>cd LibreChat

C:\LibreChat>copy .env.example .env
已复制         1 个文件。

C:\LibreChat>docker compose up
time="2024-02-20T21:17:12+08:00" level=warning msg="The \"UID\" variable is not set. Defaulting to a blank string."
time="2024-02-20T21:17:12+08:00" level=warning msg="The \"GID\" variable is not set. Defaulting to a blank string."
time="2024-02-20T21:17:12+08:00" level=warning msg="The \"UID\" variable is not set. Defaulting to a blank string."
time="2024-02-20T21:17:12+08:00" level=warning msg="The \"GID\" variable is not set. Defaulting to a blank string."
time="2024-02-20T21:17:12+08:00" level=warning msg="The \"UID\" variable is not set. Defaulting to a blank string."
time="2024-02-20T21:17:12+08:00" level=warning msg="The \"GID\" variable is not set. Defaulting to a blank string."
[+] Running 1/1
 ! api Warning                                                                                                                                                                          7.0s
[+] Building 195.1s (9/10)                                                                                                                                                    docker:default
 => [api internal] load build definition from Dockerfile                                                                                                                                0.0s
 => => transferring dockerfile: 757B                                                                                                                                                    0.0s
 => [api internal] load metadata for docker.io/library/node:18-alpine                                                                                                                   3.1s
 => [api internal] load .dockerignore                                                                                                                                                   0.0s
 => => transferring context: 265B                                                                                                                                                       0.0s
 => [api internal] load build context                                                                                                                                                   0.8s
 => => transferring context: 5.94MB                                                                                                                                                     0.7s
 => [api 1/6] FROM docker.io/library/node:18-alpine@sha256:ca9f6cb0466f9638e59e0c249d335a07c867cd50c429b5c7830dda1bed584649                                                             0.0s
 => CACHED [api 2/6] COPY . /app                                                                                                                                                        0.0s
 => CACHED [api 3/6] WORKDIR /app                                                                                                                                                       0.0s
 => CACHED [api 4/6] RUN touch .env                                                                                                                                                     0.0s
 => ERROR [api 5/6] RUN apk --no-cache add curl &&     npm ci                                                                                                                         191.0s
------
 > [api 5/6] RUN apk --no-cache add curl &&     npm ci:
0.475 fetch https://dl-cdn.alpinelinux.org/alpine/v3.19/main/x86_64/APKINDEX.tar.gz
3.458 fetch https://dl-cdn.alpinelinux.org/alpine/v3.19/community/x86_64/APKINDEX.tar.gz
4.930 (1/8) Installing ca-certificates (20230506-r0)
5.121 (2/8) Installing brotli-libs (1.1.0-r1)
5.289 (3/8) Installing c-ares (1.24.0-r1)
5.451 (4/8) Installing libunistring (1.1-r2)
5.691 (5/8) Installing libidn2 (2.3.4-r4)
5.886 (6/8) Installing nghttp2-libs (1.58.0-r0)
6.150 (7/8) Installing libcurl (8.5.0-r0)
6.309 (8/8) Installing curl (8.5.0-r0)
6.520 Executing busybox-1.36.1-r15.trigger
6.538 Executing ca-certificates-20230506-r0.trigger
6.623 OK: 15 MiB in 25 packages
190.9 npm ERR! code ECONNRESET
190.9 npm ERR! errno ECONNRESET
190.9 npm ERR! network request to https://registry.npmjs.org/rimraf/-/rimraf-5.0.5.tgz failed, reason: Client network socket disconnected before secure TLS connection was established
190.9 npm ERR! network This is a problem related to network connectivity.
190.9 npm ERR! network In most cases you are behind a proxy or have bad network settings.
190.9 npm ERR! network
190.9 npm ERR! network If you are behind a proxy, please make sure that the
190.9 npm ERR! network 'proxy' config is set properly.  See: 'npm help config'
190.9
190.9 npm ERR! A complete log of this run can be found in: /root/.npm/_logs/2024-02-20T13_17_31_796Z-debug-0.log
------
failed to solve: process "/bin/sh -c apk --no-cache add curl &&     npm ci" did not complete successfully: exit code: 1

===========================================================================
SECOND TRY
===========================================================================
time="2024-02-20T21:17:12+08:00" level=warning msg="The \"UID\" variable is not set. Defaulting to a blank string."
time="2024-02-20T21:17:12+08:00" level=warning msg="The \"GID\" variable is not set. Defaulting to a blank string."
time="2024-02-20T21:17:12+08:00" level=warning msg="The \"UID\" variable is not set. Defaulting to a blank string."
time="2024-02-20T21:17:12+08:00" level=warning msg="The \"GID\" variable is not set. Defaulting to a blank string."
time="2024-02-20T21:17:12+08:00" level=warning msg="The \"UID\" variable is not set. Defaulting to a blank string."
time="2024-02-20T21:17:12+08:00" level=warning msg="The \"GID\" variable is not set. Defaulting to a blank string."
[+] Running 1/1
 ! api Warning                                                                                                                                                                          7.0s
[+] Building 195.1s (9/10)                                                                                                                                                    docker:default
 => [api internal] load build definition from Dockerfile                                                                                                                                0.0s
 => => transferring dockerfile: 757B                                                                                                                                                    0.0s
 => [api internal] load metadata for docker.io/library/node:18-alpine                                                                                                                   3.1s
 => [api internal] load .dockerignore                                                                                                                                                   0.0s
 => => transferring context: 265B                                                                                                                                                       0.0s
 => [api internal] load build context                                                                                                                                                   0.8s
 => => transferring context: 5.94MB                                                                                                                                                     0.7s
 => [api 1/6] FROM docker.io/library/node:18-alpine@sha256:ca9f6cb0466f9638e59e0c249d335a07c867cd50c429b5c7830dda1bed584649                                                             0.0s
 => CACHED [api 2/6] COPY . /app                                                                                                                                                        0.0s
 => CACHED [api 3/6] WORKDIR /app                                                                                                                                                       0.0s
 => CACHED [api 4/6] RUN touch .env                                                                                                                                                     0.0s
 => ERROR [api 5/6] RUN apk --no-cache add curl &&     npm ci                                                                                                                         191.0s
------
 > [api 5/6] RUN apk --no-cache add curl &&     npm ci:
0.475 fetch https://dl-cdn.alpinelinux.org/alpine/v3.19/main/x86_64/APKINDEX.tar.gz
3.458 fetch https://dl-cdn.alpinelinux.org/alpine/v3.19/community/x86_64/APKINDEX.tar.gz
4.930 (1/8) Installing ca-certificates (20230506-r0)
5.121 (2/8) Installing brotli-libs (1.1.0-r1)
5.289 (3/8) Installing c-ares (1.24.0-r1)
5.451 (4/8) Installing libunistring (1.1-r2)
5.691 (5/8) Installing libidn2 (2.3.4-r4)
5.886 (6/8) Installing nghttp2-libs (1.58.0-r0)
6.150 (7/8) Installing libcurl (8.5.0-r0)
6.309 (8/8) Installing curl (8.5.0-r0)
6.520 Executing busybox-1.36.1-r15.trigger
6.538 Executing ca-certificates-20230506-r0.trigger
6.623 OK: 15 MiB in 25 packages
190.9 npm ERR! code ECONNRESET
190.9 npm ERR! errno ECONNRESET
190.9 npm ERR! network request to https://registry.npmjs.org/rimraf/-/rimraf-5.0.5.tgz failed, reason: Client network socket disconnected before secure TLS connection was established
190.9 npm ERR! network This is a problem related to network connectivity.
190.9 npm ERR! network In most cases you are behind a proxy or have bad network settings.
190.9 npm ERR! network
190.9 npm ERR! network If you are behind a proxy, please make sure that the
190.9 npm ERR! network 'proxy' config is set properly.  See: 'npm help config'
190.9
190.9 npm ERR! A complete log of this run can be found in: /root/.npm/_logs/2024-02-20T13_17_31_796Z-debug-0.log
------
failed to solve: process "/bin/sh -c apk --no-cache add curl &&     npm ci" did not complete successfully: exit code: 1

C:\LibreChat>docker compose up
time="2024-02-20T21:22:35+08:00" level=warning msg="The \"UID\" variable is not set. Defaulting to a blank string."
time="2024-02-20T21:22:35+08:00" level=warning msg="The \"GID\" variable is not set. Defaulting to a blank string."
time="2024-02-20T21:22:35+08:00" level=warning msg="The \"UID\" variable is not set. Defaulting to a blank string."
time="2024-02-20T21:22:35+08:00" level=warning msg="The \"GID\" variable is not set. Defaulting to a blank string."
time="2024-02-20T21:22:35+08:00" level=warning msg="The \"UID\" variable is not set. Defaulting to a blank string."
time="2024-02-20T21:22:35+08:00" level=warning msg="The \"GID\" variable is not set. Defaulting to a blank string."
[+] Running 1/1
 ! api Warning                                                                                                                                                                          3.5s
[+] Building 327.0s (9/10)                                                                                                                                                    docker:default
 => [api internal] load build definition from Dockerfile                                                                                                                                0.0s
 => => transferring dockerfile: 757B                                                                                                                                                    0.0s
 => [api internal] load metadata for docker.io/library/node:18-alpine                                                                                                                   2.0s
 => [api internal] load .dockerignore                                                                                                                                                   0.0s
 => => transferring context: 265B                                                                                                                                                       0.0s
 => [api internal] load build context                                                                                                                                                   0.1s
 => => transferring context: 63.93kB                                                                                                                                                    0.1s
 => [api 1/6] FROM docker.io/library/node:18-alpine@sha256:ca9f6cb0466f9638e59e0c249d335a07c867cd50c429b5c7830dda1bed584649                                                             0.0s
 => CACHED [api 2/6] COPY . /app                                                                                                                                                        0.0s
 => CACHED [api 3/6] WORKDIR /app                                                                                                                                                       0.0s
 => CACHED [api 4/6] RUN touch .env                                                                                                                                                     0.0s
 => ERROR [api 5/6] RUN apk --no-cache add curl &&     npm ci                                                                                                                         324.7s
------
 > [api 5/6] RUN apk --no-cache add curl &&     npm ci:
0.341 fetch https://dl-cdn.alpinelinux.org/alpine/v3.19/main/x86_64/APKINDEX.tar.gz
158.1 085B8E7B0E7F0000:error:0A000126:SSL routines:ssl3_read_n:unexpected eof while reading:ssl/record/rec_layer_s3.c:303:
158.2 WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.19/main: Permission denied
158.2 fetch https://dl-cdn.alpinelinux.org/alpine/v3.19/community/x86_64/APKINDEX.tar.gz
324.6 085B8E7B0E7F0000:error:0A000126:SSL routines:ssl3_read_n:unexpected eof while reading:ssl/record/rec_layer_s3.c:303:
324.6 WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.19/community: Permission denied
324.6 ERROR: unable to select packages:
324.6   curl (no such package):
324.6     required by: world[curl]
------
failed to solve: process "/bin/sh -c apk --no-cache add curl &&     npm ci" did not complete successfully: exit code: 1

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
It's a network issue with one or more of the packages, probably due to the bandwidth and/or your location. You can try using one of the pre-built packages here instead: https://github.com/danny-avila?tab=packages&repo_name=LibreChat

See this guide for configuration: https://docs.librechat.ai/install/configuration/docker_override.html

**Reply by XMLC2024** [reply]
Thanks
I'll go try it

**Reply by danny-avila** [reply]
Also try searching the discussions posts as this has been experienced before:
- https://github.com/danny-avila/LibreChat/discussions/1287
- https://github.com/danny-avila/LibreChat/discussions/1662
- https://github.com/danny-avila/LibreChat/discussions/1533

**Reply by XMLC2024** [reply]
I am back again
I followed the link you sent and it failed again, but the error message changed.

docker-compose.override.yml:
version: '3.4'

services:
  api:
    volumes:
      - ./librechat.yaml:/app/librechat.yaml
    image: ghcr.io/danny-avila/librechat-dev:latest

  mongodb:
    image: mongo:4.4.18
error message:
 => ERROR [api 5/6] RUN apk --no-cache add curl &&     npm ci                                                                                                                         246.2s
------
 > [api 5/6] RUN apk --no-cache add curl &&     npm ci:
0.454 fetch https://dl-cdn.alpinelinux.org/alpine/v3.19/main/x86_64/APKINDEX.tar.gz
2.977 fetch https://dl-cdn.alpinelinux.org/alpine/v3.19/community/x86_64/APKINDEX.tar.gz
4.028 (1/8) Installing ca-certificates (20230506-r0)
4.195 (2/8) Installing brotli-libs (1.1.0-r1)
4.365 (3/8) Installing c-ares (1.24.0-r1)
4.499 (4/8) Installing libunistring (1.1-r2)
4.706 (5/8) Installing libidn2 (2.3.4-r4)
4.848 (6/8) Installing nghttp2-libs (1.58.0-r0)
4.988 (7/8) Installing libcurl (8.5.0-r0)
5.183 (8/8) Installing curl (8.5.0-r0)
5.590 Executing busybox-1.36.1-r15.trigger
5.608 Executing ca-certificates-20230506-r0.trigger
5.681 OK: 15 MiB in 25 packages
144.3 npm WARN deprecated abab@2.0.6: Use your platform's native atob() and btoa() methods instead
148.1 npm WARN deprecated domexception@4.0.0: Use your platform's native DOMException instead
245.5 npm notice
245.5 npm notice New minor version of npm available! 10.2.4 -> 10.4.0
245.5 npm notice Changelog: <https://github.com/npm/cli/releases/tag/v10.4.0>
245.5 npm notice Run `npm install -g npm@10.4.0` to update!
245.5 npm notice
245.5 npm ERR! code ERR_TLS_CERT_ALTNAME_INVALID
245.5 npm ERR! errno ERR_TLS_CERT_ALTNAME_INVALID
245.5 npm ERR! request to https://registry.npmjs.org/zod-to-json-schema/-/zod-to-json-schema-3.22.4.tgz failed, reason: Hostname/IP does not match certificate's altnames: Host: registry.npmjs.org. is not in the cert's altnames: DNS:qq789.shop
245.5
245.5 npm ERR! A complete log of this run can be found in: /root/.npm/_logs/2024-02-20T13_41_28_582Z-debug-0.log
------
failed to solve: process "/bin/sh -c apk --no-cache add curl &&     npm ci" did not complete successfully: exit code: 1

**Reply by XMLC2024** [reply]
Feel sorry
I forgot to turn off the VPN
This is a new error message
------
 > [api 5/6] RUN apk --no-cache add curl &&     npm ci:
0.983 fetch https://dl-cdn.alpinelinux.org/alpine/v3.19/main/x86_64/APKINDEX.tar.gz
20.80 082B26D9557F0000:error:0A000126:SSL routines:ssl3_read_n:unexpected eof while reading:ssl/record/rec_layer_s3.c:303:
20.83 fetch https://dl-cdn.alpinelinux.org/alpine/v3.19/community/x86_64/APKINDEX.tar.gz
20.83 WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.19/main: Permission denied
40.76 082B26D9557F0000:error:0A000126:SSL routines:ssl3_read_n:unexpected eof while reading:ssl/record/rec_layer_s3.c:303:
40.77 WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.19/community: Permission denied
40.77 ERROR: unable to select packages:
40.78   curl (no such package):
40.78     required by: world[curl]
------
